src/nomad_simulations/schema_packages/properties/spectral_profile.py

```diff
@@ -101,12 +101,45 @@ def resolve_pdos_name(self, logger: 'BoundLogger') -> Optional[str]:
             )
             return None
 
-        # Resolve the `name` from the `entity_ref`
         name = None
+        # If the reference is to an AtomsState, use its atom_definition_ref
         if isinstance(self.entity_ref, AtomsState):
-            name = f'atom {self.entity_ref.chemical_symbol}'
+            if (
+                self.entity_ref.atom_definition_ref
+                and self.entity_ref.atom_definition_ref.chemical_symbol
+            ):
+                name = f'atom {self.entity_ref.atom_definition_ref.chemical_symbol}'
+            else:
+                logger.warning(
+                    'AtomsState missing atom_definition_ref or chemical_symbol.'
+                )
+                return None
+        # If the reference is to an OrbitalsState, try to use its parent's chemical symbol.
         elif isinstance(self.entity_ref, OrbitalsState):
-            name = f'orbital {self.entity_ref.l_quantum_symbol}{self.entity_ref.ml_quantum_symbol} {self.entity_ref.m_parent.chemical_symbol}'
+            parent_symbol = None
+            if (
+                hasattr(self.entity_ref, 'm_parent')
+                and self.entity_ref.m_parent is not None
+            ):
+                # Try to extract chemical symbol from parent's atom_definition_ref
+                if (
+                    hasattr(self.entity_ref.m_parent, 'atom_definition_ref')
+                    and self.entity_ref.m_parent.atom_definition_ref
+                ):
+                    parent_symbol = (
+                        self.entity_ref.m_parent.atom_definition_ref.chemical_symbol
+                    )
+            # Fallback: if the OrbitalsState itself has an atom_definition_ref (if defined), use it.
+            if parent_symbol is None and hasattr(
+                self.entity_ref, 'atom_definition_ref'
+            ):
+                parent_symbol = self.entity_ref.atom_definition_ref.chemical_symbol
+            if parent_symbol is None:
+                logger.warning(
+                    'Could not resolve chemical symbol for OrbitalsState reference.'
+                )
+                return None
```

Comment:
1. You can split this more efficiently (from a performance pov) into an indented if-else block.
2. Note that deserialization may also fail and return a `str` instead. @TLCFEM is this anticipated behavior in that case?